lib/ball.js

```diff
@@ -0,0 +1,61 @@
+function Ball(options, canvas){
+  this.canvas = canvas;
+  this.ctx = this.canvas.getContext('2d');
+  this.r = options.r > 0 && options.r < 20 && options.r || 8; // can be fixed all the time?
+  this.x = options.x > 0 && options.x < canvas.width && options.x || 0 + this.r;
+  this.y = options.y > 0 && options.y < canvas.height && options.y || 0 + this.r;
+  this.sAngle = 0
+  this.eAngle = 2 * Math.PI
+}
+
+Ball.prototype.draw = function() {
```

Comment:
If we did `Ball.prototype.draw = function (context) { … }`, then we wouldn't needed to store the context.